### ISSUE TITLE
Waypoint block problems in server game

### ISSUE THREAD

**Issue by SkySem** [KAMKEEL/CustomNPC-Plus#63]
We are getting a server crash for the game when two players are in range of the "customnpcs:npcWaypoint" block when it is for a quest.
If one person enters the zone of operation of this block with the right quest for the trigger of this block, everything happens in the order of things. But if two players fall into this zone to trigger this block, the server gets a crash.

Crash [*click*](https://pastebin.com/HywpVUJe)

* Minecraft: 1.7.10
* Server core: KCaldron
* CustomNPC+: v1.6-beta2
I do not deny the possibility of a problem with another mod, however I do not see a single hint in the crash log to even suggest which mod caused this error...

**Comment by KAMKEEL** [comment]
Should be resolved in 1.6 Release

**Comment by KAMKEEL** [comment]
Reopen if not fixed in `1.6`

**Comment by SkySem** [comment]
everything happens the same unfortunately

**Comment by SkySem** [comment]
```
_java.util.ConcurrentModificationException
	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1442)
	at java.util.HashMap$ValueIterator.next(HashMap.java:1471)
	at noppes.npcs.blocks.tiles.TileWaypoint.func_145845_h(TileWaypoint.java:49)
	at net.minecraft.world.World.func_72939_s(World.java:2529)
	at net.minecraft.world.WorldServer.func_72939_s(WorldServer.java:633)
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:954)
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:431)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:809)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:669)
	at java.lang.Thread.run(Thread.java:748)_
```

**Comment by KAMKEEL** [comment]
Will be fixed in 1.6.2
